main.js

```diff
@@ -1865,6 +1961,35 @@ function _select_Gateway() { // Check for Gateway used to
         },
     });
 
+    // expand recursive tasklist
+    for (var professionIdx in definedTask) {

```

Comment:
Shouldn't scare me, if I know how large this array is :D
Are you scared to: 
<pre>for (var i=0; i<13; i++)
  for (var j=0; j<20; j++)</pre>